spec/ordinary-and-exotic-object-behaviours.html

```diff
@@ -0,0 +1,322 @@
+        <p>When the [[GetOwnProperty]] internal method of a Record exotic object _R_ is called with property key _P_, the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. If Type(_P_) is a Symbol, return *undefined*.
+          1. Let _getResult_ be ! RecordGet(_R_, _P_).
+          1. If _getResult_ is ~empty~, return *undefined*.
+          1. Return the PropertyDescriptor { [[Value]]: _getResult_, [[Writable]]: *false*, [[Enumerable]]: *true*, [[Configurable]]: *false* }.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-record-exotic-objects-defineownproperty-p-desc">
+        <h1>[[DefineOwnProperty]] ( _P_, _Desc_ )</h1>
+        <p>When the [[DefineOwnProperty]] internal method of a Record exotic object _R_ is called with property key _P_, and Property Descriptor _Desc_ the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. If _Desc_.[[Writable]] is present and has value *true*, return *false*.
+          1. If _Desc_.[[Enumerable]] is present and has value *false*, return *false*.
+          1. If _Desc_.[[Configurable]] is present and has value *true*, return *false*.
+          1. Let _current_ be ! RecordGet(_R_, _P_).
+          1. If _current_ is ~empty~ return *false*.
+          1. Assert: IsDataDescriptor(_Desc_) is *true*.
+          1. If _Desc_.[[Value]] is present, return ! SameValue(_Desc_.[[Value]], _current_).
+          1. Return *false*.
+        </emu-alg>
+      </emu-clause>
+      
+      <emu-clause id="sec-record-exotic-objects-hasproperty-p">
+        <h1>[[HasProperty]] ( _P_ )</h1>
+        <p>When the [[HasProperty]] internal method of a Record exotic object _R_ is called with property key _P_, the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. If Type(_P_) is a Symbol, return *false*.
+          1. Return ! RecordHasProperty(_R_, _P_).
+        </emu-alg>
+      </emu-clause>
+            
+      <emu-clause id="sec-record-exotic-objects-get-p-receiver">
+        <h1>[[Get]] ( _P_, _Receiver_ )</h1>
+        <p>When the [[Get]] internal method of a Record exotic object _R_ is called with property key _P_ and ECMAScript language value _Receiver_, the following steps are taken:</p>
+        <emu-alg>
+          1. Assert: IsPropertyKey(_P_) is *true*.
+          1. If Type(_P_) is a Symbol, return *undefined*.
+          1. Let _value_ be ! RecordGet( _R_, _P_ ).
+          1. If _value_ is ~empty~, return *undefined*.
+          1. Return _value_
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-record-exotic-objects-set-p-receiver">
+        <h1>[[Set]] ( _P_, _Receiver_ )</h1>
+        <p>When the [[Set]] internal method of a Record exotic object _R_ is called with Property Key _P_ and ECMAScript language value _Receiver_, the following steps are taken:</p>
```

Comment:
I think I'm gonna allow symbols to be set in that version of the spec but still not support string props just yet until we solve #137

Reply:
We discussed this in committee and I explained why I think we should continue to prohibit defining symbols on Record wrappers. There is no way we could let the definition of these symbols be reflected on the underlying record--it would be superficially on the wrapper. I don't think we should encourage these wrappers to be used in a stateful way, and I don't think it would be very useful (e.g., equality doesn't work). You can always put a Proxy around a record wrapper if you want to add other behaviors like this.

Reply:
Ok, I'm reverting that change, will discuss further in #137. Either way is easy to change in the spec so I don't want to block this PR because of it.